meson.build

```diff
@@ -0,0 +1,1795 @@
+}
+vaapi_x11 += {'use': vaapi['use'] and x11['use'] and vaapi_x11['deps'].found()}
+if vaapi_x11['use'] and get_option('gpl')
+    dependencies += vaapi_x11['deps']
+    features += vaapi_x11['name']
+    sources += files('video/out/vo_vaapi.c')
+endif
+
+vaapi_x_egl = {
+    'name': 'vaapi-x-egl',
+    'use': vaapi_x11['use'] and egl_x11.allowed(),
+}
+if vaapi_x_egl['use']
+    features += vaapi_x_egl['name']
+endif
+
+vaapi_egl = {
+    'name': 'vaapi-egl',
+    'use': vaapi_x_egl['use'] or vaapi_wayland['use'] or vaapi_drm['use'],
+}
+if vaapi_egl['use']
+    dependencies += [vaapi_wayland['deps'], vaapi_drm['deps']]
+    features += vaapi_egl['name']
+    sources += files('video/out/hwdec/hwdec_vaapi_gl.c')
+endif
+
+vaapi_vulkan = {
+    'name': 'vaapi-vulkan',
+    'use': vaapi['use'] and libplacebo.found() and vulkan.found(),
+}
+if vaapi_vulkan['use']
+    features += vaapi_vulkan['name']
+    sources += files('video/out/hwdec/hwdec_vaapi_vk.c')
+endif
+
+if vaapi_egl['use'] or vaapi_vulkan['use']
+    sources += files('video/out/hwdec/hwdec_vaapi.c')
+endif
+
+vdpau = dependency('vdpau', version: '>= 0.2', required: get_option('vdpau'))
+if vdpau.found() and x11['use']
+    dependencies += vdpau
+    features += 'vdpau'
+    sources += files('video/filter/vf_vdpaupp.c',
+                     'video/out/vo_vdpau.c',
+                     'video/vdpau.c',
+                     'video/vdpau_mixer.c')
+endif
+if vdpau.found() and gl_x11.allowed()
+    features += 'vdpau-gl-x11'
+    sources += files('video/out/opengl/hwdec_vdpau.c')
+endif
+
+videotoolbox_gl = get_option('videotoolbox-gl').require(
+    gl_cocoa.allowed() or ios_gl,
+    error_message: 'gl-cocoa nor ios-gl could be found!',
+)
+if videotoolbox_gl.allowed()
+    features += 'videobox-gl'
+    sources += files('video/out/opengl/hwdec_osx.c')
+endif
+
+
+# macOS features
+macos_sdk_version_py = find_program(join_paths(source_root, 'TOOLS',
+                                    'macos-sdk-version.py'))
+macos_sdk_info = run_command(macos_sdk_version_py).stdout().split(',')
+macos_sdk_path = macos_sdk_info[0].strip()
+macos_sdk_version = macos_sdk_info[1]
+
+if macos_sdk_version != '0.0'
+    message('Detected macOS SDK ' + macos_sdk_version)
+endif
+
+macos_10_11_features = get_option('macos-10-11-features').require(
+    macos_sdk_version.version_compare('>=10.11'),
+    error_message: 'a suitable macos sdk version could not be found!'
+)
+
+
+macos_10_12_2_features = get_option('macos-10-12-2-features').require(
+    macos_sdk_version.version_compare('>=10.12.2'),
+    error_message: 'a suitable macos sdk version could not be found!'
+)
+
+macos_10_14_features = get_option('macos-10-14-features').require(
+    macos_sdk_version.version_compare('>=10.14'),
+    error_message: 'a suitable macos sdk version could not be found!'
+)
+
+swift = get_option('swift-build').require(
+    darwin,
+    error_message: 'the os is not darwin!',
```

Comment:
doesn't it make sense to check for the supported swift and sdk version here, similar to all the other checks and the original code https://github.com/mpv-player/mpv/blob/9d5d9b24240efe98cf99bbda2cb5280b025506d8/wscript#L185-L189

right now the version check is in "generated/osdep/meson.build", which seems inconsistent?

Reply:
Yup definitely makes sense.